xcresultparser/XCResultFormatter.swift

```diff
@@ -32,7 +33,8 @@ struct XCResultFormatter {
     // MARK: - Initializer
     
     init?(with url: URL,
-          formatter: XCResultFormatting
+          formatter: XCResultFormatting,
+          targets: [String]?
```

Comment:
The way you have implemented it it would require the consumer to specify the targets. (In main.swift the default value is an empty list). I would rather like it to behave as before and only change behaviour, if the extra parameter for 'targets' is given. Also get rid of the optional, by defaulting to all targets. I also would recommend a Set of Strings here.
```
init?(with url: URL,
    formatter: XCResultFormatting,
    coverageTargets: Set<String> = []
) {
...
...
self.coverageTargets = codeCoverage?.targets(filteredBy: coverageTargets) ?? []
```
and then create an extension on CodeCoverage like so:
```
extension CodeCoverage {
    func targets(filteredBy filter: Set<String>) -> Set<String> {
        guard !filter.isEmpty else {
            return Set(targets.map { $0.name })
        }
        let filtered = targets.map { $0.name }
            .filter { thisTarget in
                guard let stripped = thisTarget.split(separator: ".").first else { return true }
                return filter.contains(String(stripped))
            }
        return Set(filtered)
    }
}
```
now we can simply do a contains in the loop through the coverage data in line 211:
```
for target in codeCoverage.targets {
    guard coverageTargets.contains(target.name) else { continue }
```

Reply:
Good ideas. I have kept the coverageTargets as an array until the processing in the CodeCoverage  extension. It seemed as the logical place for doing the Array to Set transformation.